### PR TITLE
Copy //runtime/dart/utils from Topaz into the engine.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Build rules and include paths have been updated as necessary. No other functional change.

### CODE REVIEW

File: shell/platform/fuchsia/runtime/dart/utils/files.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// Copyright 2019 The Fuchsia Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef TOPAZ_RUNTIME_DART_UTILS_FILES_H_
```
Comment (jason-simmons) [inline shell/platform/fuchsia/runtime/dart/utils/files.h]: Should the include guards be updated to not mention Topaz?

Comment (chinmaygarde) [reply shell/platform/fuchsia/runtime/dart/utils/files.h]: Updated.

**Comment by chinmaygarde** [comment]
Header guards have been updated. PTAL.

File: shell/platform/fuchsia/runtime/dart/utils/BUILD.gn [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "$fuchsia_sdk_root/pkg:memfs",
+    "$fuchsia_sdk_root/pkg:syslog",
+    "$fuchsia_sdk_root/pkg:zx",
+    "$fuchsia_sdk_root/pkg/lib/sys/cpp",
+    "$fuchsia_sdk_root/pkg/lib/vfs/cpp",
+    "//third_party/tonic",
+  ]
+
+  public_configs = [ ":utils_config" ]
+}
```
Comment (zanderso) [inline shell/platform/fuchsia/runtime/dart/utils/BUILD.gn]: This drops the package rule for run_vmservice_object_tests. We need to ensure that that tests continues to run after the migration.

Comment (chinmaygarde) [reply shell/platform/fuchsia/runtime/dart/utils/BUILD.gn]: That rule refers to the non-existent (yet) template definitions in `//build/package.gni`. I was going to build up those rules in subsequent packages. This rule just works builds what is necessary for existing targets.

May I add those rules in subsequent patches after I port the template definitions?

Comment (zanderso) [reply shell/platform/fuchsia/runtime/dart/utils/BUILD.gn]: sgtm

**Comment by chinmaygarde** [comment]
Presubmits have been failing because of 502 on DEPS.

**Comment by chinmaygarde** [comment]
I have addressed all issues, lgtm?

**Pull request merged**